src/app/beer_garden/plugin.py

```diff
@@ -0,0 +1,141 @@
+from datetime import datetime
+from typing import Sequence
+
+from brewtils.models import Instance, System
+
+import beer_garden.db.api as db
+import beer_garden.queue.api as queue
+
+
+def register(system: System):
+    """Register a System
+
+    Create a queue if required and update the database.
+
+    Args:
+        system:
+
+    Returns:
+
+    """
+    instance = db.query_unique(Instance, id=instance_id)
+    system = db.query_unique(System, instances__contains=instance)
+
+    logger.info(
+        "Initializing instance %s[%s]-%s", system.name, instance.name, system.version
+    )
+
+    routing_words = [system.name, system.version, instance.name]
+
+    request_queue_name = get_routing_key(*routing_words)
+    request_queue = queue.create(
+        request_queue_name,
+        [request_queue_name],
+        durable=True,
+        arguments={"x-max-priority": 1},
+    )
+
+    suffix = [random.choice(string.ascii_lowercase + string.digits) for _ in range(10)]
+    routing_words.append("".join(suffix))
+
+    admin_keys = get_routing_keys(*routing_words, is_admin=True)
+    admin_queue = queue.create(admin_keys[-1], admin_keys, durable=True)
+
+    amq_config = beer_garden.config.get("amq")
+    connection = {
+        "host": beer_garden.config.get("publish_hostname"),
+        "port": amq_config.connections.message.port,
+        "user": amq_config.connections.message.user,
+        "password": amq_config.connections.message.password,
+        "virtual_host": amq_config.virtual_host,
+        "ssl": {"enabled": amq_config.connections.message.ssl.enabled},
+    }
+
+    instance.status = "INITIALIZING"
+    instance.status_info = {"heartbeat": datetime.utcnow()}
+    instance.queue_type = "rabbitmq"
+    instance.queue_info = {
+        "admin": {"name": admin_queue.name},
+        "request": {"name": request_queue.name},
+        "connection": connection,
+    }
+    instance = db.update(instance)
```

Comment:
Would there be a CON in moving the DB call into the queue.rabbit.create() function? Then we would only have to call queue.create(). A lot of this code is redundant between the two functions.